ruby/ql/test/query-tests/security/decompression-api/decompression_api.rb

```diff
@@ -0,0 +1,42 @@
+class TestController < ActionController::Base
+    def unsafe_zlib_unzip
+        Zlib::Inflate.inflate(params[:fname])
+    end
+
+    def safe_zlib_unzip
+        Zlib::Inflate.inflate("testfile.gz")
+    end
+
+
+    DECOMPRESSION_LIB = Zlib
+    def unsafe_zlib_unzip_const
+        DECOMPRESSION_LIB::Inflate.inflate(params[:fname])
+    end
+
+    def unsafe_zlib_unzip
+        Zip::File.open(params[:fname]) do |zip_file|
+            zip_file.each do |entry|
+                entry.extract(entry.name)
+            end
+        end
+    end
+
+    def safe_zlib_unzip
+        Zip::Entry.extract("testfile.gz")
+    end
+
+    def sanitized_zlib_unzip
+        fname = params[:fname]
+        if fname == "safe_file.gz"
```

Comment:
I think I was able to get this test fixed where it wouldn't result in a hit from my query, but I'm still a little confused why it wasn't working before. Looking at the implementation for `StringConstCompare`, it looks like it should be able to see the comparison regardless of which side of the assignment operator it's on, but that didn't seem to be the case. Is it because the tainted value must first be assigned to `fname` in order for a DataFlow node to exist?